ocs_ci/ocs/upgrade.py

```
@@ -0,0 +1,537 @@
+            "minCount", constants.MIN_NB_ENDPOINT_COUNT_POST_DEPLOYMENT
+        )
+
+        if not (min_endpoints <= endpoint_count <= max_endpoints):
+            raise ValueError(
+                f"Endpoint pod count {endpoint_count} not in allowed range [{min_endpoints}, {max_endpoints}]"
+            )
+
+        if len(noobaa_pod_obj) != len(expected_noobaa_pods):
+            raise ValueError(
+                f"Expected noobaa pods: {expected_noobaa_pods} do not match actual noobaa pods: {noobaa_pod_names}"
+            )
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)